src/backends/gr.jl

```diff
@@ -657,6 +666,9 @@ function _update_min_padding!(sp::Subplot{GRBackend})
     if sp[:yaxis][:guide] != ""
         leftpad += 4mm
     end
+    if sp[:colorbar_title] != ""
+        rightpad += 4mm
```

Comment:
Is it required?  I added it in the patch just to follow how other guides work (and ask this question) but it seems that this includes extra unnecessary space in the plot.

Reply:
If it doesn't look right then don't include it - you might post an example?

Reply:
I don't really understand why the right padding should be increased by a fixed amount. I think it depends on the title's text extent.

Reply:
Ok, now I see. The title is displayed vertically, so it only depends on the text height.

Reply:
@jheinen Yes, I thought it makes sense but incrementing the right padding introduces extra space which is a bit unexpected.

@mkborregaard So maybe `gr_colorbar_ratio` is too large?  But what I tried was to set the bottom of `colorbar_title` at the right edge of the figure (= `viewport_plotarea[2] + gr_colorbar_ratio`):

https://github.com/JuliaPlots/Plots.jl/blob/2acd088226bde5a5ce58358369dfa804a9da8731/src/backends/gr.jl#L489-L493

So if `gr_colorbar_ratio` is too large, I think there has to be a space between the colorbar and title, not between the right edge of the figure and the title.  But this is not the case and I'm puzzled.

Reply:
OK. So I think I found the source of my confusion.  I was thinking that `viewport_plotarea[2] + gr_colorbar_ratio` would be the edge of entire image (i.e., anything right to it would not be rendered in (say) png file).  But that's not the case, right?  My current understanding is that it is the right edge of area for the "subplots/axes" (not sure what's the technical word... plot area?) and the guides (title/xlabel/ylabel) have to be located outside of it.  If this is correct, using `TEXT_VALIGN_BOTTOM` was wrong since that would put the colorbar title inside the edge.  So I changed it to `TEXT_VALIGN_TOP`:

https://github.com/JuliaPlots/Plots.jl/blob/b56238089cf3985da11574497659045ec881b896/src/backends/gr.jl#L489-L493

With this change (and still keeping `rightpad += 4mm`) I now have:

![fig](https://user-images.githubusercontent.com/29282/44188662-a2857900-a0d3-11e8-8844-172714fb5aef.png)

I think this is the right direction.  Previous two examples seem to have colorbar title that is too close to the colorbar tick labels.

Reply:
Yes, this should be correct.